frontend/src/Pages/HomePage.js

```diff
@@ -20,19 +41,109 @@ export default function HomePage() {
         socket.emit('join-lobby', { roomCode: code, nickname });
     };
 
+    const validateNickname = (nickname) => {
+        let validCheck = true;
+        setNickname(nickname);
+        if (nickname === '') {
+            validCheck = false;
+            setnonEmpty(invalidNickname);
+            setTickNonempty(false);
+        } else {
+            setnonEmpty(validNickname);
+            setTickNonempty(true);
+        }
+        if (nickname.includes(' ')) {
+            validCheck = false;
+            setnoSpaces(invalidNickname);
+            setTickNoSpaces(false);
+        } else {
+            setnoSpaces(validNickname);
+            setTickNoSpaces(true);
+        }
+        if (nickname.length > 10) {
+            validCheck = false;
+            setLessThanTen(invalidNickname);
+            setTickLessThanTen(false);
+        } else {
+            setLessThanTen(validNickname);
+            setTickLessThanTen(true);
+        }
+        if (validCheck) {
+            setjoinDisabled(false);
+        } else {
+            setjoinDisabled(true);
+        }
+    };
+
     return (
         <div>
             <div className={styles.container}>
                 <h1>MAFIA</h1>
-                <TextField
-                    color="secondary"
-                    id="nickname"
-                    label="Enter Nickname"
-                    autoComplete="off"
-                    variant="outlined"
-                    value={nickname}
-                    onChange={(e) => setNickname(e.target.value)}
-                ></TextField>
+                <ClickAwayListener onClickAway={handleTooltipClose}>
+                    <div>
+                        <Tooltip
+                            PopperProps={{
+                                disablePortal: true,
+                            }}
+                            onClose={handleTooltipClose}
+                            open={open}
+                            disableFocusListener
```

Comment:
any reason why these are disabled?

Reply:
These listeners disabled so that the tooltip won't go away when user moves mouse away from component. Tooltip only goes away when clicking outside component.